components/event-handler-notification/org.wso2.carbon.identity.event.handler.notification/src/main/java/org/wso2/carbon/identity/event/handler/notification/util/NotificationUtil.java

```diff
@@ -299,6 +352,131 @@ public static void deployPublisher(EventPublisherConfiguration eventPublisherCon
+        OMElement fallbackElem = configParser.getConfigElement(
+                NotificationConstants.EmailNotification.ORGANIZATION_LEVEL_EMAIL_BRANDING_FALLBACKS_ELEM);
+        if (fallbackElem == null) {
+            return Collections.emptyMap();
+        }
+
+        Iterator iterator = fallbackElem.getChildrenWithLocalName(
+                NotificationConstants.EmailNotification.ORGANIZATION_LEVEL_EMAIL_BRANDING_FALLBACK_ELEM);
+        if (iterator == null) {
+            return Collections.emptyMap();
+        }
+
+        Map<String, String> fallbackMap = new HashMap<>();
+        while (iterator.hasNext()) {
+            OMElement omElement = (OMElement) iterator.next();
+            if (omElement != null) {
+                String key = omElement.getAttributeValue(
+                        new QName(NotificationConstants.EmailNotification.ORGANIZATION_LEVEL_EMAIL_BRANDING_FALLBACK_KEY_ATTRIBUTE));
+                String value = omElement.getText();
+                fallbackMap.put(key, value);
+            }
+        }
+        return fallbackMap;
+    }
+
+    /**
+     * Retrive branding value by the placeholder.
+     *
+     * @param key                   placeholder in the email template.
+     * @param brandingPreferences   list of branding preferences.
+     * @param brandingFallbacks     default branding values.
+     * @return map of default organization branding.
+     */
+    public static String getBrandingPreference(String key, JsonNode brandingPreferences, Map<String, String> brandingFallbacks) {
+
+        String value = null;
+        boolean brandingIsEnabled = (brandingPreferences != null)
+                && brandingPreferences.at(NotificationConstants.EmailNotification.BRANDING_PREFERENCES_IS_ENABLED_PATH).asBoolean();
+        String theme = brandingIsEnabled
+                ? brandingPreferences.at("/theme/activeTheme").asText()
+                : NotificationConstants.EmailNotification.BRANDING_PREFERENCES_LIGHT_THEME;
+
+        switch (key) {
+            case "organization.logo.img" :
+                value = (brandingIsEnabled && StringUtils.isNotBlank(
+                            getBrandingPreferenceByTheme(brandingPreferences, theme, "/images/logo/imgURL")))
+                        ? getBrandingPreferenceByTheme(brandingPreferences, theme, "/images/logo/imgURL")
+                        : brandingFallbacks.get("logoUrl");
+                break;
+            case "organization.logo.altText" :
+                value = (brandingIsEnabled && StringUtils.isNotBlank(
+                            getBrandingPreferenceByTheme(brandingPreferences, theme, "/images/logo/altText")))
+                        ? getBrandingPreferenceByTheme(brandingPreferences, theme, "/images/logo/altText")
+                        : StringUtils.EMPTY;
+                break;
+            case "organization.copyright.text" :
+                value = (brandingIsEnabled && StringUtils.isNotBlank(
```

Comment:
Shall we check whether we can combine the logic in this case?

Reply:
Fixed with 46e40d5d5cd8ab8fe019a9915da6e5e9ae3693f1